CheckDuplicateContactsHelper.cls

```diff
@@ -11,11 +11,18 @@
  **************************************************************************************************/
 public with sharing class CheckDuplicateContactsHelper {
 	public void validateContact(List <Contact> contactlist){
-		List<Contact> contacts = new List<Contact>();
+		List<Contact> contacts = [Select Phone, Email from Contact];
```

Comment:
Here you will end up `SOQL` limit.
This logic needs to update. Time complexity for this logic is more coz you are using loop into loop instead of that you can go for map and improve the performance for this code.